src/librustc_ast/ast.rs

```diff
@@ -2153,7 +2153,7 @@ impl FnRetTy {
 /// Module declaration.
 ///
 /// E.g., `mod foo;` or `mod foo { .. }`.
-#[derive(Clone, RustcEncodable, RustcDecodable, Debug)]
+#[derive(Clone, RustcEncodable, RustcDecodable, Debug, Default)]
```

Comment:
This derived `Default` is pretty suspicious.
If it's `inline: false`, why doesn't expand tries to load it from file again?
Can it use some real span instead of the default `DUMMY_SP`?

Reply:
>  If it's `inline: false`, why doesn't expand tries to load it from file again?

Hmm.. Are you assuming somehow that `flat_map_item` in expand is called twice on the same `mod foo;`? If that's the case, we could turn `inline: bool` to `inline: InlineState` where `enum IsInline { No, Yes, YesExpanded }`.

>  Can it use some real span instead of the default `DUMMY_SP`?

Where would the `Span` come from? The documentation for `inner` states that:
```rust
    /// For `mod foo;`, the inner span ranges from the first token
    /// to the last token in the external file.
```

When the file hasn't been loaded, an inner span seems like a meaningless notion, so a dummy seems appropriate? (We can also exclude the illegal state by amending `InlineState` to `enum InlineState { No(Span), Yes, YesExpanded(Span) }`.)

Reply:
> This derived `Default` is pretty suspicious.

Well I just did this to reduce code; the change should preserve existing semantics, although `Default` might be an invitation to misuse?

Reply:
>although Default might be an invitation to misuse?

This was my primary concern, the `Default` seems to only accidentally match what we need during recovery.

Reply:
Anyway, not a big deal.